src/components/organisms/UiMenu/UiMenu.vue

```diff
@@ -62,139 +53,179 @@ export interface MenuProps {
    */
   items?: (string | MenuRenderItem)[];
   /**
-   * Use this props to allow using arrows to navigation.
+   * Use this props to allow using arrows instead tabs for navigation UiMenuItems.
    */
   enableKeyboardNavigation?: boolean;
+  /**
+   * Use this props to pass refs to UiMenuItems.
+   */
+  itemsTemplateRefs?: ComponentInstance<typeof UiMenuItem>[] | [];
 }
 export type MenuAttrsProps = DefineAttrsProps<MenuProps, ListAttrsProps>;
-export interface MenuItem {
-  $el: HTMLLIElement,
-  tabindex: Ref<number | null>
-}
 
 const props = withDefaults(defineProps<MenuProps>(), {
   items: () => ([]),
-  enableKeyboardNavigation: false,
+  enableKeyboardNavigation: true,
+  itemsTemplateRefs: () => ([]),
 });
-const menuItemAttrs = ({
-  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
-  name, label, ...itemAttrs
-}: MenuRenderItem) => itemAttrs;
-const itemsToRender = computed<MenuRenderItem[]>(() => (props.items.map((item, key) => {
+provide('enableKeyboardNavigation', props.enableKeyboardNavigation);
+const itemsToRender = computed<MenuRenderItem[]>(() => (props.items.map((item, index) => {
   if (typeof item === 'string') {
     return {
-      name: `menu-item-${key}`,
+      name: `menu-item-${index}`,
       label: item,
     };
   }
   return {
-    name: `menu-item-${key}`,
+    name: `menu-item-${index}`,
     ...item,
   };
 })));
-const menuItems = ref<MenuItem[] | []>([]);
-provide('menuItems', menuItems);
-const mItems = computed(() => (menuItems.value));
+const disabledKeyboardNavigation = computed(() => (!props.enableKeyboardNavigation));
+const menuItemsTemplateRefs = ref<InstanceType<typeof UiMenuItem>[]>([]);
+watch(props.itemsTemplateRefs, () => {
+  menuItemsTemplateRefs.value = props.itemsTemplateRefs;
+});
+if (props.enableKeyboardNavigation
+    && props.itemsTemplateRefs
+    && props.itemsTemplateRefs.length < 1) {
+  console.warn('@infermedica/component-library: use itemsTemplateRefs to pass UiMenuItems template refs.');
+}
 const {
-  firstMenuItem,
-  lastMenuItem,
-  nextMenuItem,
-  prevMenuItem,
-  selectedMenuItem,
-  focusedMenuItem,
-} = useMenuItems(mItems);
-const handleMenuKeydown = async ({ key }: KeyboardEvent) => {
-  if (!props.enableKeyboardNavigation) return;
+  focusedElement,
+  firstElement,
+  initialElement,
+  lastElement,
+  selectedElement,
+  nextElement,
+  prevElement,
+} = useArrowNavigation(menuItemsTemplateRefs);
+const setNegativeTabindexForNonInitialMenuItems = async () => {
+  menuItemsTemplateRefs.value.forEach((item) => {
+    if (item === initialElement.value) {
+      return;
+    }
+    item.tabindex = -1;
```

Comment:
<img width="697" alt="image" src="https://github.com/infermedica/component-library/assets/37744371/1a74eaa6-bba0-4c21-b52d-76e4e9fbd3b1">

Reply:
to fix this issue I disable this rule in eslint